### PR TITLE
Notif on update

### PR DESCRIPTION
**Pull request by Adrien-COSTANDI**

### CODE REVIEW

File: src/front/src/conf/backendController.ts [hunk 1]
```diff
@@ -153,3 +153,42 @@ export async function createNewElement<T>(path: string, data: any): Promise<Requ
...
+            body: JSON.stringify(data)
+        } :
+        {
+            method: method,
+        }
+    let response = await fetch(path, request)
+
+    if (response.ok) {
+        let result = (await response.json()) as T
+        if(result){
```
Comment (tvelu77) [inline src/front/src/conf/backendController.ts]:
```suggestion
        if (result) {
```

File: src/front/src/conf/backendController.ts [hunk 2]
```diff
@@ -153,3 +153,42 @@ export async function createNewElement<T>(path: string, data: any): Promise<Requ
...
+            return {
+                succes: result
+            }
+        }
+        console.error("Unexpected response type received: " + path, response)
+    } else {
+        let error = (await response.json()) as ErrorMessage
+        if(error){
```
Comment (tvelu77) [inline src/front/src/conf/backendController.ts]:
```suggestion
        if (error) {
```

File: src/front/src/conf/backendController.ts [hunk 3]
```diff
@@ -153,3 +153,42 @@ export async function createNewElement<T>(path: string, data: any): Promise<Requ
...
+            return {
+                error: error
+            }
+        }
+        console.error("UUpdate element request failed: " + path, response)
```
Comment (tvelu77) [inline src/front/src/conf/backendController.ts]:
```suggestion
        console.error("Update element request failed: " + path, response)
```

File: src/front/src/conf/backendController.ts [hunk 4]
```diff
@@ -121,30 +121,30 @@ export type ErrorMessage = {
...
             headers: {
                 "Content-Type": "application/json",
             },
             body: JSON.stringify(data)
         })
-    if (request.ok) {
-        let result = (await request.json()) as T
+    if (response.ok) {
+        let result = (await response.json()) as T
         if(result){
```
Comment (tvelu77) [inline src/front/src/conf/backendController.ts]:
```suggestion
        if (result) {
```

File: src/front/src/pages/home/configuration/CreateConfig.tsx [hunk 5]
```diff
@@ -89,22 +88,48 @@ function CreateConfig(props: { id?: number }) {
...
+                //         });
+                //         navigate("/home/configuration");
+                //     } else {
+                //         wsManager.emitNotification({
+                //             type: "ERROR",
+                //             title: "Erreur ",
+                //             content: "La configuration n'a pas pu être modifiée."
+                //         })
+                //     }
+                // })
```
Comment (tvelu77) [inline src/front/src/pages/home/configuration/CreateConfig.tsx]: Maybe delete this ?

File: src/front/src/pages/home/firmware/CreateFirmware.tsx [hunk 6]
```diff
@@ -149,7 +149,46 @@ export default function CreateFirmware(props: { id?: number, data?: CreateFirmwa
...
+                constructor: firmware.constructor,
+                typesAllowed: typesArray,
+            })
+                .then(firmwareRequest => {
+                    if (firmwareRequest.succes) {
+                        let firmware = firmwareRequest.succes
+                        wsManager.emitNotification({
+                            type: "SUCCESS",
+                            title: firmware.version + " | " + firmware.constructor + " ",
+                            content: "La firmware a été modifié."
```
Comment (tvelu77) [inline src/front/src/pages/home/firmware/CreateFirmware.tsx]:
```suggestion
                            content: "Le firmware a été modifié."
```

File: src/main/java/fr/uge/chargepointconfiguration/firmware/FirmwareService.java [hunk 7]
```diff
@@ -121,7 +120,12 @@ private void checkAlreadyExisting(CreateFirmwareDto createFirmwareDto) {
    * @return A firmware updated with its information.
    */
   public Firmware update(int id, CreateFirmwareDto createFirmwareDto) {
-    checkAlreadyExisting(createFirmwareDto);
+    var firm = firmwareRepository.findByUrl(createFirmwareDto.url());
+    // I can have the same URL if it's me
```
Comment (tvelu77) [inline src/main/java/fr/uge/chargepointconfiguration/firmware/FirmwareService.java]:
```suggestion
    // Firm can have the same URL if it's itself
```

**Review by tvelu77 — CHANGES_REQUESTED** [review]

File: src/front/src/pages/home/configuration/CreateConfig.tsx (hunk 5 above)
Comment (Adrien-COSTANDI) [reply src/front/src/pages/home/configuration/CreateConfig.tsx]: > Maybe delete this ?

no

Comment (Adrien-COSTANDI) [inline src/front/src/pages/home/configuration/CreateConfig.tsx]:
```suggestion
```

**Review by tvelu77 — APPROVED** [review]

**Pull request merged**